### PR TITLE
20160623 102800 desktop pyserial

### PR DESCRIPTION
**Pull request by andySigler**
These diffs revert the backend python code to be run on a downloadable app on Mac and Windows. It accomplishes the following:
1. Removes Raspberry Pi specific scripts and functionality
2. Replaces ser2net module with pyserial, for communicating over a Mac or Windows USB-Serial port
3. Adds automatic Smoothieboard detect-connect-disconnect feature, as well as reporting USB connection status to frontend

### CODE REVIEW

File: backend/file_io.py [hunk 1]
```diff
@@ -39,7 +39,7 @@ def log(*msg):
         tstamp = datetime.datetime.now()
         try:
             logfile = None
-            logfile = open('logfile.txt',"a")
+            logfile = open('otone_data/logfile.txt',"a")
```
Comment (SimplyAhmazing) [inline backend/file_io.py:42]: What does this line do when running as an executable on windows/mac? where's the file created..

Comment (andySigler) [inline backend/file_io.py:42]: we'll need to do some testing as an executables, but the otone_data folder is created by the code to store machine-specific data, so I figured the logfile should also go in there

File: backend/publisher.py [hunk 2]
```diff
@@ -124,7 +125,7 @@ def on_finish(self):     #called from planner/theQueue
         """Publish status and move on to next instruction step
         """
         if debug == True: FileIO.log('publisher.on_finish called')
-        self.send_message('status',{'string':'Robot stopped','color':'black'})
+        # self.send_message('status',{'string':'Robot stopped','color':'black'})
```
Comment (SimplyAhmazing) [inline backend/publisher.py]: commented code is dead code and shouldn't be part of the repo

File: backend/file_io.py (hunk 1 above)
Comment (SimplyAhmazing) [inline backend/file_io.py:42]: I'm planning on having all this file FileIO stuff removed and if anything we can use python's build in logger.

Comment (andySigler) [inline backend/file_io.py:42]: rolling in the logging module in the following branch:

https://github.com/OpenTrons/otone_backend/tree/201606270123100-desktop-logging

**Pull request merged**